src/transform.js

```diff
@@ -647,31 +646,84 @@ function untransformObject(schema, className, mongoObject, isNestedObject = fals
 
```

Comment:
I went with something simpler. Moving this logic all to the mongo adapter will be a better time to fix it up.